shell/src/test/java/org/apache/accumulo/shell/commands/FateCommandTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.shell.commands;
+
+import static org.apache.accumulo.fate.zookeeper.ServiceLock.ServiceLockPath;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import org.apache.accumulo.fate.AdminUtil;
+import org.apache.accumulo.fate.ReadOnlyRepo;
+import org.apache.accumulo.fate.ReadOnlyTStore;
+import org.apache.accumulo.fate.ZooStore;
+import org.apache.accumulo.fate.zookeeper.ZooReaderWriter;
+import org.easymock.EasyMock;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FateCommandTest {
+  private static ZooReaderWriter zk;
+  private static ServiceLockPath managerLockPath;
+
+  @BeforeClass
+  public static void setup() {
+    zk = EasyMock.createMock(ZooReaderWriter.class);
+    managerLockPath = EasyMock.createMock(ServiceLockPath.class);
+  }
+
+  @Test
+  public void testFailTx() throws Exception {
+    ZooStore<FateCommand> zs = EasyMock.createMock(ZooStore.class);
+    String tidStr = "12345";
+    long tid = Long.parseLong(tidStr, 16);
+    EasyMock.expect(zs.getStatus(tid)).andReturn(ReadOnlyTStore.TStatus.NEW).anyTimes();
+    zs.reserve(tid);
+    EasyMock.expectLastCall().once();
+    zs.setStatus(tid, ReadOnlyTStore.TStatus.FAILED_IN_PROGRESS);
+    EasyMock.expectLastCall().once();
+    zs.unreserve(tid, 0);
+    EasyMock.expectLastCall().once();
+
+    TestHelper helper = new TestHelper(true);
+
+    replay(zs);
```

Comment:
I don't see a `verify` call anywhere. It's good practice: `create mocks -> enter replay mode -> verify mock objects`

Reply:
Good catch. Fixed in af358ef1bc0153873959ba821b67d0877206ead8